flink-core/src/main/java/org/apache/flink/configuration/HighAvailabilityOptions.java

```diff
@@ -87,6 +87,14 @@
 			.withDeprecatedKeys("recovery.jobmanager.port")
 			.withDescription("Optional port (range) used by the job manager in high-availability mode.");
 
+	/**
+	 * Optional port (range) used by the job manager in high-availability mode.
+	 */
+	public static final ConfigOption<String> HA_JOB_MANAGER_METRIC_QUERY_PORT_RANGE =
```

Comment:
Why did you introduce these configuration options? Ideally we don't need these things at all, because the metric query service is an internal service and Flink should be able to figure out automatically on which port it runs.

Reply:
I was thinking that this ports should be specified by users, but seems like we can use any free port to run the service.